app/lib/server/functions/saveUser.js

```diff
@@ -146,8 +147,19 @@ function validateUserEditing(userId, userData) {
 
 	const canEditOtherUserInfo = hasPermission(userId, 'edit-other-user-info');
 	const canEditOtherUserPassword = hasPermission(userId, 'edit-other-user-password');
+	const user = Users.findOneById(userData._id);
 
-	if (userData.roles && !hasPermission(userId, 'assign-roles')) {
+	const isEditingUserRoles = (previousRoles, newRoles) => {
+		if (!Array.isArray(previousRoles) || !Array.isArray(newRoles) || previousRoles.length !== newRoles.length) {
+			return true;
+		}
+		previousRoles.sort();
+		newRoles.sort();
+		return !previousRoles.every((role, index) => role === newRoles[index]);
+	};
```

Comment:
I think using the underscore is the cleanest way to go, you should test it, but I thinks something like this will work

`const  isEditingUserRoles  = (previousRoles, newRoles) => !_.isEqual(_.sortBy(previousRoles), _.sortBy(newRoles))`

It will sort both arrays and compare if the contain the same elements

Reply:
Much cleaner and it works, thanks 😄